src/pkg/fault/fault.go

```diff
@@ -435,68 +379,21 @@ func (pec printableErrCore) Values() []string {
 // Local aggregator
 // ---------------------------------------------------------------------------
 
-// Local constructs a new local bus to handle error aggregation in a
-// constrained scope.  Local busses shouldn't be passed down  to other
-// funcs, and the function that spawned the local bus should always
-// return `local.Failure()` to ensure that hard failures are propagated
-// back upstream.
-func (e *Bus) Local() *localBus {
-	return &localBus{
-		mu:  &sync.Mutex{},
-		bus: e,
-	}
-}
-
-type localBus struct {
-	mu      *sync.Mutex
-	bus     *Bus
-	current error
-}
-
-func (e *localBus) AddRecoverable(ctx context.Context, err error) {
-	if err == nil {
-		return
-	}
-
-	e.mu.Lock()
-	defer e.mu.Unlock()
-
-	if e.current == nil && e.bus.failFast {
-		e.current = err
+// Local constructs a new bus with a local reference to handle error aggregation
+// in a constrained scope.  This allows the caller to review recoverable errors and
+// failures within only the current codespace, as opposed to the global set of errors.
+// The function that spawned the local bus should always return `bus.Failure()` to
+// ensure that hard failures are propagated back upstream.
+func (e *Bus) Local() *Bus {
+	parent := e
+
+	// always use the root bus reference, if e is not the root.
+	if e.parent != nil {
+		e = e.parent
```

Comment:
I guess both are fine, but I feel like `parent = e.parent` conveys the information better.